src/btree/bt_sync_obsolete.c

```diff
@@ -291,3 +290,324 @@ __wt_sync_obsolete_cleanup(WT_SESSION_IMPL *session, WT_REF *parent)
 
     return (0);
 }
+
+/*
+ * __checkpoint_cleanup_run_chk --
+ *     Check to decide if the checkpoint cleanup should continue running.
+ */
+static bool
+__checkpoint_cleanup_run_chk(WT_SESSION_IMPL *session)
+{
+    return (FLD_ISSET(S2C(session)->server_flags, WT_CONN_SERVER_CHECKPOINT_CLENAUP));
+}
+
+/*
+ * __checkpoint_cleanup_page_skip --
+ *     Return if checkpoint cleanup requires we read this page.
+ */
+static int
+__checkpoint_cleanup_page_skip(
+  WT_SESSION_IMPL *session, WT_REF *ref, void *context, bool visible_all, bool *skipp)
+{
+    WT_ADDR_COPY addr;
+
+    WT_UNUSED(context);
+    WT_UNUSED(visible_all);
+
+    *skipp = false; /* Default to reading */
+
+    /*
+     * Skip deleted pages as they are no longer required for the checkpoint. The checkpoint never
+     * needs to review the content of those pages - if they should be included in the checkpoint the
+     * existing page on disk contains the right information and will be linked into the checkpoint
+     * as the internal tree structure is built.
+     */
+    if (ref->state == WT_REF_DELETED) {
+        *skipp = true;
+        return (0);
+    }
+
+    /* If the page is in-memory, we want to look at it. */
+    if (ref->state != WT_REF_DISK)
+        return (0);
+
+    /*
+     * Reading any page that is not in the cache will increase the cache size. Perform a set of
+     * checks to verify the cache can handle it.
+     */
+    if (__wt_cache_aggressive(session) || __wt_cache_full(session) || __wt_cache_stuck(session) ||
+      __wt_eviction_needed(session, false, false, NULL)) {
+        *skipp = true;
+        return (0);
+    }
+
+    /* Don't read pages into cache during startup or shutdown phase. */
```

Comment:
Should we only start this after startup and shut down this before shutdown? So there is no need for this check?

Reply:
Nice observation. We are already starting and stopping after recovery and before shutdown; this code is no longer needed. removed it.